scripts/test/shared.py

```diff
@@ -362,6 +366,8 @@ def get_tests(test_dir, extensions=[]):
         tests += glob.glob(os.path.join(test_dir, '*'))
     for ext in extensions:
         tests += glob.glob(os.path.join(test_dir, '*' + ext))
+    if options.test_name_filter:
+        tests = list(filter(lambda n: n.find(options.test_name_filter) >= 0, tests))
```

Comment:
Can you use list comprehension instead here?

```
tests = [t for t in tests if t.find(options.test_name_filter) >= 0]
```

Even if you don't use list comprehension isn't the `list()` on the outside redundant?

Reply:
`filter` returns an iterator, not a list, apparently.

Reply:
not going to open a new PR to change it into a comprehension if you don't mind..

Reply:
No worries, my feedback crossed paths with landing.